src/com/ConnectionInfoPublisher.cpp

```diff
@@ -0,0 +1,93 @@
+#include "ConnectionInfoPublisher.hpp"
+#include <chrono>
+#include <thread>
+#include <boost/uuid/name_generator.hpp>
+#include <boost/uuid/string_generator.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/filesystem.hpp>
+
+ConnectionInfoPublisher::ConnectionInfoPublisher(std::string acceptorName,
+                                                 std::string requesterName,
+                                                 int rank,
+                                                 std::string addressDirectory)
+  :
+  acceptorName(acceptorName),
+  requesterName(requesterName),
+  rank(rank),
+  addressDirectory(addressDirectory)
+{}
+  
+
+ConnectionInfoPublisher::ConnectionInfoPublisher(std::string acceptorName,
+                                                 std::string requesterName,
+                                                 std::string addressDirectory)
+  :
+  acceptorName(acceptorName),
+  requesterName(requesterName),
+  addressDirectory(addressDirectory)
+{}
+
+
+std::string ConnectionInfoPublisher::getFilename()
+{
+  using namespace boost::filesystem;
+  
+  constexpr int firstLevelLen = 2;
+  boost::uuids::string_generator ns_gen;
+  auto ns = ns_gen("af7ce8f2-a9ee-46cb-38ee-71c318aa3580"); // md5 hash of precice.org as namespace
+  
+  boost::uuids::name_generator gen{ns};
+  std::string s = acceptorName + requesterName + std::to_string(rank);
+  std::string hash = boost::uuids::to_string(gen(s));
+  hash.erase(std::remove(hash.begin(), hash.end(), '-'), hash.end());
+  
+  path p = path(addressDirectory)
+    / path(".precice")
+    / path(hash.substr(0, firstLevelLen))
+    / hash.substr(firstLevelLen);
+  
+  return p.string();  
+}
+
+
+std::string ConnectionInfoReader::read()
```

Comment:
As strange as it sounds, this call is semantically `const` wrt the object.
```suggestion
std::string ConnectionInfoReader::read() const
```

Considering the name of this class, this should probably also parse the the endpoint and return the ip-port pair.

Reply:
The connection info is not always an IP-Port pair, e.g. in case of MPI ports.